packages/contracts/test/core/acl/integration-tests/shared-component.ts

```diff
@@ -0,0 +1,199 @@
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // Create id-gated object associated with dao1
+      const tx = await testComponent.newObject(dao1.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      await expect(
+        testComponent.callStatic.idGatedAction(0)
+      ).to.be.revertedWith(expectedACLAuthError);
+    });
+
+    it('reverts if the role name is wrong', async () => {
+      // Deploy oracle and set allowed id
+      const allowedId = 1;
+
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grant signers[0] permission to do id gated actions with the deployed oracle
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ethers.utils.id('WRONG_ROLE'),
+        oracle.address
+      );
+
+      // Create id-gated object associated with dao1
+      const tx = await testComponent.newObject(dao1.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      await expect(
+        testComponent.callStatic.idGatedAction(0)
+      ).to.be.revertedWith(expectedACLAuthError);
+    });
+
+    it('reverts if the permission is set in the wrong DAO', async () => {
+      // Deploy oracle and set allowed id
+      const allowedId = 0;
+
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grant signers[0] the permission to do id gated actions with the deployed oracle
+      dao2.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // Create id-gated object associated with dao1
+      const tx = await testComponent.newObject(dao1.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      await expect(
+        testComponent.callStatic.idGatedAction(0)
+      ).to.be.revertedWith(expectedACLAuthError);
+    });
+
+    it('reverts if the object belongs to the wrong DAO', async () => {
+      // Deploy oracle and set allowed id
+      const allowedId = 0;
+
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grant signer[0] permission to do id gated actions with the deployed oracle
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // Create id-gated object associated with dao1
+      const tx = await testComponent.newObject(dao2.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      await expect(
+        testComponent.callStatic.idGatedAction(0)
```

Comment:
```suggestion
        testComponent.callStatic.idGatedAction(allowedId)
```

Reply:
In the latest version. Let's do a re-review.

Reply:
I still see `testComponent.callStatic.idGatedAction(0)` on the codebase ¿?

https://github.com/aragon/core/pull/45/files/57ffe6e02c2635f403b613622696156bc0d06bb5#diff-364aed46e75e9eb52249674bc287b9fe7ed9ee67e44c89b4f9a41901cb9b1d1eR195